src/main/java/com/microsoft/graph/serializer/DefaultSerializer.java

```diff
@@ -108,7 +105,8 @@ public DefaultSerializer(@Nonnull final ILogger logger) {
 			// If there is a derived class, try to get it and deserialize to it
 			T jo = jsonObject;
 			if (rawElement.isJsonObject()) {
-				final Class<?> derivedClass = this.getDerivedClass(rawObject, clazz);
+                final DerivedClassIdentifier derivedClassIdentifier = new DerivedClassIdentifier(logger);
```

Comment:
can we have this being a property instead assigned by the constructor? this will avoid assigning once instance for each call of the method.